src/Utilities/copy.jl

```diff
@@ -1,3 +1,69 @@
+# This file contains default implementations for the `MOI.copy_to` function that can be used by a model.
+
+"""
+    automatic_copy_to(dest::MOI.ModelLike, src::MOI.ModelLike;
+                      copy_names::Bool=true)
+
+Use [`Utilities.supports_incremental_copy`](@ref) and
+[`Utilities.supports_allocate_load`](@ref) to automatically choose between
+[`Utilities.default_copy_to`](@ref) or [`Utilities.allocate_load`](@ref) to
+apply the copy operation.
+"""
+function automatic_copy_to(dest::MOI.ModelLike, src::MOI.ModelLike;
+                           copy_names::Bool=true)
+    if supports_incremental_copy(dest, copy_names)
+        default_copy_to(dest, src, copy_names)
+    elseif supports_allocate_load(dest, copy_names)
+        allocate_load(dest, src, copy_names)
+    else
+        error("Model $(typeof(dest)) does not support copy",
+              copy_names ? " with names" : "", ".")
+    end
+end
+
+"""
+    supports_incremental_copy(model::ModelLike, copy_names::Bool)
+
+Return a `Bool` indicating whether the model `model` supports
+[`default_copy_to(model, src, copy_names=copy_names)`](@ref) if all the
+attributes set to `src` and constraints added to `src` are supported by `model`.
+
+This function can be used to determine whether a model can be loaded into
+`model` incrementally or whether it should be cached and copied at once instead.
+This is used by JuMP to determine whether to add a cache or not.
+If this function returns false whatever the value of `copy_names` is then
```

Comment:
Could this paragraph on what happens for different return values be formatted as a table or bullet points? I'm having trouble keeping all the possible conditions straight.